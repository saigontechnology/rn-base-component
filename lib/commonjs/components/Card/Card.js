"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _native = _interopRequireDefault(require("styled-components/native"));
var _metrics = require("../../helpers/metrics");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const Card = _ref => {
  let {
    onPress,
    style,
    children
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(CardWrapper, {
    onPress: onPress,
    activeOpacity: onPress ? _metrics.activeOpacity.low : _metrics.activeOpacity.none,
    style: style,
    testID: 'card'
  }, children);
};
const CardWrapper = _native.default.TouchableOpacity(props => {
  var _props$theme, _props$theme2;
  return {
    padding: props === null || props === void 0 || (_props$theme = props.theme) === null || _props$theme === void 0 || (_props$theme = _props$theme.spacing) === null || _props$theme === void 0 ? void 0 : _props$theme.slim,
    borderRadius: _metrics.metrics.borderRadius,
    backgroundColor: props === null || props === void 0 || (_props$theme2 = props.theme) === null || _props$theme2 === void 0 || (_props$theme2 = _props$theme2.colors) === null || _props$theme2 === void 0 ? void 0 : _props$theme2.cardBackground
  };
});
var _default = exports.default = Card;
//# sourceMappingURL=Card.js.map