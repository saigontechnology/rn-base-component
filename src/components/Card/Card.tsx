import React, {PropsWithChildren} from 'react'
import type {StyleProp, ViewStyle} from 'react-native'
import styled from 'styled-components/native'
import {activeOpacity, metrics} from '../../helpers/metrics'
import type {ITheme} from '../../theme'

export interface CardProps extends PropsWithChildren {
  /**
   * handle press on card
   */
  onPress?: () => void

  /**
   * style for card
   */
  style?: StyleProp<ViewStyle> | Array<StyleProp<ViewStyle>>
}

export const Card: React.FC<CardProps> = ({onPress, style, children}) => (
  <CardWrapper
    onPress={onPress}
    activeOpacity={onPress ? activeOpacity.low : activeOpacity.none}
    style={style}
    testID={'card'}>
    {children}
  </CardWrapper>
)

const CardWrapper = styled.TouchableOpacity((props: {theme: ITheme}) => ({
  padding: props?.theme?.spacing?.slim,
  borderRadius: metrics.borderRadius,
  backgroundColor: props?.theme?.colors?.cardBackground,
}))
