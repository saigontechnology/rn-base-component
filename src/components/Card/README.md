# Card Component

A flexible and reusable card container component for React Native applications that provides a consistent layout foundation with optional touch interactions.

## Features

- ðŸŽ¨ **Themed Styling** - Consistent styling that integrates with the design system
- ðŸ‘† **Touch Interactions** - Optional onPress functionality with appropriate feedback
- ðŸ”§ **Flexible Layout** - Accepts any children for maximum flexibility
- ðŸ“± **Responsive Design** - Adapts to different screen sizes
- â™¿ **Accessibility Ready** - Built-in accessibility features
- ðŸŽ¯ **TypeScript Support** - Full type safety and IntelliSense

## Installation

```bash
npm install rn-base-component
# or
yarn add rn-base-component
```

## Basic Usage

```tsx
import React from 'react'
import {Card} from 'rn-base-component'
import {Text, View} from 'react-native'

export default function App() {
  return (
    <Card>
      <Text>This is a simple card</Text>
    </Card>
  )
}
```

## Interactive Card

```tsx
<Card onPress={() => console.log('Card pressed!')}>
  <Text>Tap this card</Text>
</Card>
```

## Advanced Usage

### Custom Styled Card

```tsx
;<Card style={styles.customCard}>
  <Text>Card with custom shadow</Text>
</Card>

const styles = StyleSheet.create({
  customCard: {
    margin: 16,
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
})
```

### Card with Complex Content

```tsx
;<Card onPress={() => navigateToDetails()}>
  <View style={styles.profileContent}>
    <Image source={{uri: 'https://example.com/avatar.jpg'}} style={styles.avatar} />
    <View style={styles.profileInfo}>
      <Text style={styles.profileName}>John Doe</Text>
      <Text style={styles.profileRole}>Software Engineer</Text>
      <Text style={styles.profileActivity}>Last active 2 hours ago</Text>
    </View>
    <Icon name="chevron-right" size={20} color="#ccc" />
  </View>
</Card>

const styles = StyleSheet.create({
  profileContent: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  avatar: {
    width: 50,
    height: 50,
    borderRadius: 25,
  },
  profileInfo: {
    marginLeft: 12,
    flex: 1,
  },
  profileName: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  profileRole: {
    color: '#666',
    marginTop: 2,
  },
  profileActivity: {
    color: '#999',
    fontSize: 12,
    marginTop: 4,
  },
})
```

### List of Cards

```tsx
;<ScrollView>
  {items.map((item, index) => (
    <Card key={index} style={styles.listCard} onPress={() => handleItemPress(item)}>
      <Text style={styles.cardTitle}>{item.title}</Text>
      <Text style={styles.cardDescription}>{item.description}</Text>
    </Card>
  ))}
</ScrollView>

const styles = StyleSheet.create({
  listCard: {
    marginHorizontal: 16,
    marginBottom: 12,
  },
  cardTitle: {
    fontSize: 18,
    fontWeight: '600',
  },
  cardDescription: {
    color: '#666',
    marginTop: 4,
  },
})
```

## API Reference

### CardProps

| Prop              | Type                                                  | Default     | Description                                         |
| ----------------- | ----------------------------------------------------- | ----------- | --------------------------------------------------- |
| `onPress`         | `() => void`                                          | `undefined` | Callback function when card is pressed              |
| `style`           | `StyleProp<ViewStyle> \| Array<StyleProp<ViewStyle>>` | `undefined` | Custom styles for the card                          |
| `children`        | `React.ReactNode`                                     | Required    | Content to be rendered inside the card              |
| `padding`         | `number`                                              | Theme       | Padding inside the card (overrides theme default)   |
| `borderRadius`    | `number`                                              | Theme       | Border radius of the card (overrides theme default) |
| `backgroundColor` | `string`                                              | Theme       | Background color of the card (overrides theme)      |

### Default Styling

The Card component comes with default styling from the theme:

- **Padding**: `theme.components.Card.padding` (internal padding)
- **Border Radius**: `theme.components.Card.borderRadius`
- **Background Color**: `theme.components.Card.backgroundColor`
- **Touch Opacity**: Low opacity when pressable, none when not pressable

## Layout Patterns

### Profile Card

```tsx
;<Card style={styles.profileCard}>
  <View style={styles.profileContainer}>
    <Image source={{uri: userAvatar}} style={styles.profileAvatar} />
    <Text style={styles.profileUserName}>{userName}</Text>
    <Text style={styles.profileUserRole}>{userRole}</Text>
    <TouchableOpacity style={styles.followButton}>
      <Text style={styles.followButtonText}>Follow</Text>
    </TouchableOpacity>
  </View>
</Card>

const styles = StyleSheet.create({
  profileCard: {
    margin: 16,
  },
  profileContainer: {
    alignItems: 'center',
    padding: 20,
  },
  profileAvatar: {
    width: 80,
    height: 80,
    borderRadius: 40,
  },
  profileUserName: {
    fontSize: 20,
    fontWeight: 'bold',
    marginTop: 12,
  },
  profileUserRole: {
    color: '#666',
    marginTop: 4,
  },
  followButton: {
    marginTop: 16,
    backgroundColor: '#007AFF',
    paddingHorizontal: 24,
    paddingVertical: 8,
    borderRadius: 20,
  },
  followButtonText: {
    color: 'white',
    fontWeight: '600',
  },
})
```

### Statistics Card

```tsx
;<Card style={styles.statsCard}>
  <Text style={styles.statsValue}>{value}</Text>
  <Text style={styles.statsLabel}>{label}</Text>
  <Text style={styles.statsChange}>â†— +12% from last month</Text>
</Card>

const styles = StyleSheet.create({
  statsCard: {
    margin: 8,
    flex: 1,
  },
  statsValue: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#007AFF',
  },
  statsLabel: {
    color: '#666',
    marginTop: 4,
  },
  statsChange: {
    color: '#00AA00',
    fontSize: 12,
    marginTop: 8,
  },
})
```

### Article Card

```tsx
;<Card onPress={() => openArticle(article.id)} style={styles.articleCard}>
  <Image source={{uri: article.image}} style={styles.articleImage} />
  <View style={styles.articleContent}>
    <Text style={styles.articleTitle}>{article.title}</Text>
    <Text style={styles.articleExcerpt}>{article.excerpt}</Text>
    <View style={styles.articleFooter}>
      <Text style={styles.articleDate}>{article.date}</Text>
      <Text style={styles.readMoreText}>Read More</Text>
    </View>
  </View>
</Card>

const styles = StyleSheet.create({
  articleCard: {
    margin: 16,
  },
  articleImage: {
    width: '100%',
    height: 200,
    borderRadius: 8,
  },
  articleContent: {
    padding: 16,
  },
  articleTitle: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  articleExcerpt: {
    color: '#666',
    marginTop: 8,
    lineHeight: 20,
  },
  articleFooter: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: 12,
  },
  articleDate: {
    color: '#999',
    fontSize: 12,
  },
  readMoreText: {
    color: '#007AFF',
    fontSize: 14,
  },
})
```

### Settings Item Card

```tsx
;<Card onPress={() => navigateToSetting(setting.id)}>
  <View style={styles.settingItem}>
    <View style={[styles.settingIcon, {backgroundColor: setting.color}]}>
      <Icon name={setting.icon} size={20} color="white" />
    </View>
    <View style={styles.settingContent}>
      <Text style={styles.settingTitle}>{setting.title}</Text>
      <Text style={styles.settingDescription}>{setting.description}</Text>
    </View>
    <Icon name="chevron-right" size={16} color="#ccc" />
  </View>
</Card>

const styles = StyleSheet.create({
  settingItem: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 8,
  },
  settingIcon: {
    width: 40,
    height: 40,
    borderRadius: 8,
    alignItems: 'center',
    justifyContent: 'center',
  },
  settingContent: {
    flex: 1,
    marginLeft: 12,
  },
  settingTitle: {
    fontSize: 16,
    fontWeight: '600',
  },
  settingDescription: {
    color: '#666',
    fontSize: 14,
  },
})
```

## Theme Integration

The Card component integrates with the theme system and can be customized via theme configuration:

### Theme Configuration

```tsx
import {extendTheme} from 'rn-base-component'

const customTheme = extendTheme({
  components: {
    Card: {
      padding: 20, // Custom padding
      borderRadius: 12, // Custom border radius
      backgroundColor: '#F8F9FA', // Custom background color
    },
  },
})
```

### Using Theme Values

```tsx
// Uses theme defaults
<Card>
  <Text>This card uses theme styling</Text>
</Card>

// Override specific theme values
<Card
  padding={24}
  backgroundColor="#FFFFFF"
  borderRadius={16}
>
  <Text>This card overrides theme values</Text>
</Card>
```

### Default Theme Values

```tsx
// Default Card theme configuration
CardTheme: {
  padding: 16,              // theme.spacing.slim
  borderRadius: 8,          // metrics.borderRadius
  backgroundColor: '#FFFFFF' // theme.colors.cardBackground
}
```

## Styling Guidelines

### Elevation and Shadows

```tsx
// iOS shadow
<Card style={styles.iosShadow}>

// Android elevation
<Card style={styles.androidElevation}>

// Cross-platform shadow
<Card style={styles.crossPlatformShadow}>

const styles = StyleSheet.create({
  iosShadow: {
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.1,
    shadowRadius: 4,
  },
  androidElevation: {
    elevation: 3,
  },
  crossPlatformShadow: {
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
})
```

### Spacing

```tsx
// Card margins for list layouts
<Card style={styles.listLayout}>

// Full-width cards
<Card style={styles.fullWidth}>

// Grid cards
<Card style={styles.gridCard}>

const styles = StyleSheet.create({
  listLayout: {
    marginHorizontal: 16,
    marginBottom: 12,
  },
  fullWidth: {
    marginVertical: 8,
  },
  gridCard: {
    margin: 8,
    flex: 1,
  },
})
```

## Accessibility

### Screen Reader Support

```tsx
<Card
  onPress={handlePress}
  accessibilityRole="button"
  accessibilityLabel="User profile card for John Doe"
  accessibilityHint="Tap to view full profile">
  {/* Card content */}
</Card>
```

### Non-interactive Cards

```tsx
<Card accessibilityRole="text">{/* Static content */}</Card>
```

## Best Practices

1. **Touch Feedback** - Only add `onPress` when the card should be interactive
2. **Content Hierarchy** - Use consistent padding and spacing inside cards
3. **Visual Consistency** - Maintain consistent card styling across the app
4. **Performance** - Use `StyleSheet.create()` for static styles
5. **Accessibility** - Provide appropriate accessibility props for interactive cards
6. **Shadow Consistency** - Use consistent shadow/elevation values

## Common Use Cases

### Dashboard Cards

```tsx
;<View style={styles.dashboardContainer}>
  {metrics.map(metric => (
    <Card key={metric.id} style={styles.dashboardCard}>
      <Text style={styles.metricValue}>{metric.value}</Text>
      <Text style={styles.metricLabel}>{metric.label}</Text>
    </Card>
  ))}
</View>

const styles = StyleSheet.create({
  dashboardContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    padding: 8,
  },
  dashboardCard: {
    width: '48%',
    margin: '1%',
  },
  metricValue: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  metricLabel: {
    color: '#666',
  },
})
```

### Navigation Menu

```tsx
;<ScrollView>
  {menuItems.map(item => (
    <Card key={item.id} onPress={() => navigate(item.route)}>
      <View style={styles.menuItem}>
        <Icon name={item.icon} size={24} />
        <Text style={styles.menuItemText}>{item.title}</Text>
      </View>
    </Card>
  ))}
</ScrollView>

const styles = StyleSheet.create({
  menuItem: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  menuItemText: {
    marginLeft: 16,
    fontSize: 16,
  },
})
```

## Performance Considerations

- Cards with images should implement proper image loading and caching
- For long lists, consider using FlatList with Card components
- Avoid complex nested layouts inside cards for better performance
- Use `removeClippedSubviews` for long scrollable card lists

## Troubleshooting

### Common Issues

**Card not responding to touch**

- Ensure `onPress` prop is provided
- Check if there are overlapping touchable components
- Verify the card has sufficient size for touch interaction

**Styling not applying**

- Check style prop syntax and ensure valid style objects
- Verify theme configuration is correct
- Use style array format for multiple style objects

**Shadow not showing on Android**

- Use `elevation` property instead of shadow properties
- Ensure the card has sufficient margin for shadow visibility
