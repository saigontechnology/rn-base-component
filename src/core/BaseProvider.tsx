import React, {createContext, useCallback, useMemo, useState} from 'react'
import {ThemeProvider} from 'styled-components/native'
import {ITheme, theme as defaultTheme} from '../theme'
import type {ColorMode, IColorModeContextProps} from './color-mode/type'

export type IBaseContext = {theme: ITheme} | IColorModeContextProps
export const BaseContext = createContext<IBaseContext | null>(null)

export interface BaseProviderProps {
  children?: React.ReactNode
  theme?: ITheme
}

export const BaseProvider = ({children, theme = defaultTheme}: BaseProviderProps) => {
  const [colorModeValue, setColorModeValue] = useState(theme?.config.initialColorMode)

  const isLight = useMemo(() => colorModeValue === 'light', [colorModeValue])

  const darkTheme = useMemo(() => ({...theme, colors: {...theme.colors, ...theme.darkColors}}), [theme])

  const newTheme = useMemo(() => (isLight ? theme : darkTheme), [darkTheme, isLight, theme])

  const toggleColorMode = useCallback(() => {
    isLight ? setColorModeValue('dark') : setColorModeValue('light')
  }, [isLight])

  const setColorMode = useCallback((value: ColorMode) => {
    setColorModeValue(value)
  }, [])

  return (
    <BaseContext.Provider
      value={{
        theme: newTheme,
        colorMode: colorModeValue,
        toggleColorMode,
        setColorMode,
      }}>
      <ThemeProvider theme={newTheme}>{children}</ThemeProvider>
    </BaseContext.Provider>
  )
}
