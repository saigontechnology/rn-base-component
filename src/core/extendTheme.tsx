/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/no-explicit-any */
import {theme as defaultTheme, Theme} from './../theme'
import _ from 'lodash'

function isFunction(value: any): boolean {
  return typeof value === 'function'
}

type ThemeUtil = Theme | (Record<string, any> & {})

export function extendTheme<T extends ThemeUtil>(overrides: T, ...restOverrides: T[]) {
  function customizer(source: any, override: any) {
    if (isFunction(source)) {
      return (...args: any[]) => {
        const sourceValue = source(...args)
        const overrideValue = isFunction(override) ? override(...args) : override
        return _.mergeWith({}, sourceValue, overrideValue, customizer)
      }
    }
    return undefined
  }

  const finalOverrides = [overrides, ...restOverrides].reduce(
    (prevValue, currentValue) => _.mergeWith({}, prevValue, currentValue, customizer),
    defaultTheme,
  )

  return finalOverrides as T & Theme
}
