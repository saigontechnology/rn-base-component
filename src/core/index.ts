export * from './extendTheme'
export * from './BaseProvider'
