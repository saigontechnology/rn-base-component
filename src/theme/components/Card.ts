import type {StyleProp, ViewStyle} from 'react-native'
import {metrics} from '../../helpers'
import base from '../base'

export type CardThemeProps = {
  /**
   * Padding inside the card
   */
  padding: number
  /**
   * Border radius for the card
   */
  borderRadius: number
  /**
   * Background color of the card
   */
  backgroundColor: string
  /**
   * Custom style for the card container
   */
  style?: StyleProp<ViewStyle>
}

export const CardTheme: CardThemeProps = {
  padding: base.spacing.slim,
  borderRadius: metrics.borderRadius,
  backgroundColor: base.colors.cardBackground,
  style: undefined,
}
