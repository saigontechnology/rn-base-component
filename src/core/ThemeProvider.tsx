import React, {createContext, useCallback, useMemo, useState} from 'react'
import {IColors, ITheme, theme as defaultTheme} from 'src/theme'
import {
  ThemeProvider as ThemeProviderStyled,
  ThemeContext as ThemeContextStyled,
  ThemeProviderComponent,
} from 'styled-components'
import type {ColorMode, IColorModeContextProps} from './color-mode/type'

export const BaseContext = createContext<IColorModeContextProps | {}>({})
type AnyIfEmpty<T extends object> = keyof T extends never ? any : T
const ThemeProvider: ThemeProviderComponent<AnyIfEmpty<IColors>> = ThemeProviderStyled
export const ThemeContext: React.Context<any> = ThemeContextStyled

export interface ThemeProviderProps {
  children?: React.ReactNode
  theme?: ITheme
}

const BaseProvider = ({children, theme = defaultTheme}: ThemeProviderProps) => {
  const [colorModeValue, setColorModeValue] = useState<ColorMode>(theme?.config.initialColorMode)

  const darkColor = useMemo(() => {
    return theme.darkColors
  }, [theme.darkColors])

  const isLight = useMemo(() => {
    return colorModeValue === 'light'
  }, [colorModeValue])

  const newTheme = useMemo(() => {
    return isLight ? theme.colors : darkColor
  }, [darkColor, isLight, theme])

  const toggleColorMode = useCallback(() => {
    isLight ? setColorModeValue('dark') : setColorModeValue('light')
  }, [isLight])

  const setColorMode = useCallback((value: ColorMode) => {
    setColorModeValue(value)
  }, [])

  return (
    <BaseContext.Provider
      value={{
        colorMode: colorModeValue,
        toggleColorMode,
        setColorMode,
      }}>
      <ThemeProvider theme={newTheme}>{children}</ThemeProvider>
    </BaseContext.Provider>
  )
}

export default BaseProvider
