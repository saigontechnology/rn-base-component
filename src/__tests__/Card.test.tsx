import {fireEvent, render} from '@testing-library/react-native'
import React from 'react'
import {Text} from 'react-native'
import Card from '../components/Card/Card'
import {BaseProvider} from '../core'

describe('Card', () => {
  it('renders correctly', () => {
    const {toJSON} = render(
      <BaseProvider>
        <Card />
      </BaseProvider>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders children correctly', () => {
    const {getByText} = render(
      <BaseProvider>
        <Card>
          <Text>Test Children</Text>
        </Card>
      </BaseProvider>,
    )
    expect(getByText('Test Children')).toBeTruthy()
  })

  it('handles onPress correctly', () => {
    const mockOnPress = jest.fn()
    const {getByTestId} = render(
      <BaseProvider>
        <Card onPress={mockOnPress} />
      </BaseProvider>,
    )
    const card = getByTestId('card')
    fireEvent.press(card)
    expect(mockOnPress).toHaveBeenCalled()
  })

  it('applies background color correctly', () => {
    const {toJSON} = render(
      <BaseProvider>
        <Card />
      </BaseProvider>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
