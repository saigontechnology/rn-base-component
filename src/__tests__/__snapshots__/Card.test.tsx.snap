// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card applies background color correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderBottomLeftRadius": 8.214285714285714,
      "borderBottomRightRadius": 8.214285714285714,
      "borderTopLeftRadius": 8.214285714285714,
      "borderTopRightRadius": 8.214285714285714,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
  testID="card"
/>
`;

exports[`Card renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderBottomLeftRadius": 8.214285714285714,
      "borderBottomRightRadius": 8.214285714285714,
      "borderTopLeftRadius": 8.214285714285714,
      "borderTopRightRadius": 8.214285714285714,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
  testID="card"
/>
`;
